src/base/misc/fatfs.c

```diff
@@ -421,7 +421,9 @@ int read_boot(fatfs_t *f, unsigned char *b)
     if(bpb->v340.signature == 0x28 || bpb->v340.signature == 0x29) {
       bpb->v340.drive_number = f->drive_num;
     } else {
-      b[0x1fd] = f->drive_num;
+      if (b[0x1fd] >= 0x80 && b[0x1fd] <= 0x83) { // Only those that have it
+        b[0x1fd] = f->drive_num;
```

Comment:
Well, I don't understand.
So if there was a drive number, we put our
own - fine, but if it is not there? Where would
we write the drive number then? There seem
to be no "else" branch, so that logic can't be
valid.